### DISCUSSION TITLE
Offline Styles

### DISCUSSION THREAD

**Discussion by jordabi2** [maplibre/maplibre-gl-js#1975] [General]
I am new to Maplibre and am attempting to evaluate if this will meet the needs of a prototype project I'm working, so please excuse any "stupid" questions. 

I am working within the constraints of some resources already present on the network/local application stack, and am trying to avoid introducing too many additional third-party applications\dependencies, as it can be a painful process.  As this application will need to be used on a machine that I do not control, the web paradigm seems appropriate as I do control several other machines on the network.   This entire network is closed, meaning zero internet access.  

As other machines have PostgreSQL/PostGIS, Python, and Flask, I thought this a natural marriage. 

I curranty have a several OSM/geoJson files imported into PostGIS and am using pg_tileserve to serve them to the Maplibre application.  This works off line if I "pull the plug" once Maplibre is up and loaded and has managed to pull the styles from here:
style: 'https://api.maptiler.com/maps/streets/style.json?key=get_your_own_OpIi9ZULNHzrESv6T2vL', 

However, I have been unable to figure out how to load this, or any, style from a local resource and not require at least an initial connection to the internet.  

So, with that being said, does anybody have any examples, guidance, or thoughts?

**Comment by HarelM** [comment]
I'm not sure how meaningful my answer to you would be but I'll write it down anyway.
TL;DR it's possible, and shouldn't be too complicated.
Long answer:
The style is basically a json file, you can bundle it as part of the app you are writing (with webpack, rollup, whatever you are using).
The style is referencing some other json files which you need to get and place inside the main style file from previous paragraph.
The only thing left after that are the tile sources.
If it's geojson, you can dump that too in the big style file or have it available in the internal network.
For vector tile, you need to either have them available in the internal network, or use `addProtocol` to have them available in a different way (like stored in indexdb or something more complicated).
I hope this answers your question.
